hbase-server/src/main/java/org/apache/hadoop/hbase/tool/CanaryStatusServlet.java

```diff
@@ -0,0 +1,50 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.tool;
+
+import org.apache.hadoop.hbase.ServerName;
+import org.apache.hadoop.hbase.tmpl.tool.CanaryStatusTmpl;
+import org.apache.yetus.audience.InterfaceAudience;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.LongAdder;
+
+@InterfaceAudience.Private
+public class CanaryStatusServlet extends HttpServlet {
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+    throws ServletException, IOException {
+    CanaryTool.RegionStdOutSink sink =
+      (CanaryTool.RegionStdOutSink) getServletContext().getAttribute(
+        "sink");
+    assert sink != null : "No tool in context!";
```

Comment:
What happens when `sink == null` when running without assertions? A nasty NPE? Would be better to handle null properly.

Reply:
> What happens when `sink == null` when running without assertions? A nasty NPE? Would be better to handle null properly.

I changed it to throw a ServletException instead if `sink == null`.